TACC/Core-Styles
# fix: tup-659 all form fields too short

## Overview

Stretch **most** input fields* of **all** form types to same width.

<sup>\* Not checkboxes nor time nor date nor radio boxes.</sup>

## Related

- [TUP-659](https://jira.tacc.utexas.edu/browse/TUP-659)
- caused by #207
- inspired by #225

## Changes

1. Remove field width stretch from CMS and Login.
2. Add field width stretch to all forms.
3. Support form width stretch only if NOT a checkbox (nor date nor radio or time)*.

<sup>\* The radio and time and date fields are [unaffected for different reason](https://github.com/TACC/Core-Styles/blob/v2.21.1/src/lib/_imports/elements/form.css#L78-L85).</sup>

## Testing

1. Open all forms:
    - http://localhost:3002/components/detail/c-form--default
    - http://localhost:3002/components/detail/c-form--portal
    - http://localhost:3002/components/detail/c-form--cms *
    - http://localhost:3002/components/detail/django-cms-forms
    - http://localhost:3002/components/detail/s-form--default
    - http://localhost:3002/components/detail/s-form--portal
    - http://localhost:3002/components/detail/s-form--cms *
2. Verify widths of fields (except checkboxes, radios, date, time) are the same.

## UI

Skipped.

## Notes

We should **not** need `:--form__field--not-checkbox` (05f6f32) ever. I had added that, but then removed it in 70d3f28. All form fields, Portal or CMS, checkbox or not, should be stretched. Whether the `<input>` itself is stretched, is [managed by styles specific to certain `<input>` fields](https://github.com/TACC/Core-Styles/blob/v2.21.1/src/lib/_imports/elements/form.css#L78-L85).
